app/uk/gov/hmrc/claimvatenrolmentfrontend/services/ClaimVatEnrolmentService.scala

```diff
@@ -43,20 +43,20 @@ class ClaimVatEnrolmentService @Inject()(auditConnector: AuditConnector,
                          ec: ExecutionContext): Future[ClaimVatEnrolmentResponse] = {
     journeyService.retrieveJourneyData(journeyId, internalId).flatMap {
       journeyData =>
-        allocateEnrolmentService.allocateEnrolment(journeyData, credentialId, groupId).flatMap {
+        allocateEnrolmentService.allocateEnrolment(journeyData.get, credentialId, groupId).flatMap {
```

Comment:
Same with these .gets. You need to handle a None by returning a NotFound/BadRequest and showing the ISE error view